### PR TITLE
[reproducer] Enable configuration of NTP service on controller-0

### PR DESCRIPTION
**Pull request by rebtoor**
`controller-0` was the only vm deployed by the framework where the NTP service wasn't enabled and that led to some issues while parsing log files.

We are going to use the `timesync` role from the `linux-system-roles` collection just like we do in EDPM compute nodes in the `edpm-ansible` collection.

By default the configuration will enable all the NTP servers provided by the dhcp server, plus `pool.ntp.org` server that can be overridden declaring the `cifmw_ntp_server` variable.

Closes: https://issues.redhat.com/browse/OSPRH-9299

### CODE REVIEW

File: docs/source/usage/01_usage.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ are shared among multiple roles:
 - `cifmw_nolog`: (Bool) Toggle `no_log` value for selected tasks. Defaults to `true` (hiding those logs by default).
 - `cifmw_parent_scenario`: (String or List(String)) path to existing scenario/parameter file to inherit from.
 - `cifmw_configure_switches`: (Bool) Specifies whether switches should be configured. Computes in `reproducer.yml` playbook. Defaults to `false`.
+- `cifmw_ntp_server`: (String) Specifies an ntp server to use. Now it's only defined in `controller-0` environment. Defaults to `clock.redhat.com`.
```
Comment (tosky) [inline docs/source/usage/01_usage.md:71]: please change the default to a public server

File: roles/reproducer/tasks/configure_controller.yml [hunk 2]
```diff
@@ -468,3 +468,11 @@
       register: _sync_dep_install_result
       until: _sync_dep_install_result.finished
       retries: 20
+
+    - name: Configure the NTP service
+      vars:
+        timesync_dhcp_ntp_servers: true
+        timesync_ntp_servers:
+          - hostname: "{{ cifmw_ntp_server | default('clock.redhat.com') }}"
```
Comment (tosky) [inline roles/reproducer/tasks/configure_controller.yml:476]: the default value should be a public server

**Review by tosky — CHANGES_REQUESTED** [review]

File: docs/source/usage/01_usage.md (hunk 1 above)
Comment (rebtoor) [reply docs/source/usage/01_usage.md]: done

File: roles/reproducer/tasks/configure_controller.yml (hunk 2 above)
Comment (rebtoor) [reply roles/reproducer/tasks/configure_controller.yml]: done

File: requirements.yml [hunk 3]
```diff
@@ -47,3 +47,6 @@ collections:
     type: git
   - name: https://github.com/ovirt/ovirt-ansible-collection
     type: git
+  - name: https://github.com/linux-system-roles/timesync.git
```
Comment (lewisdenny) [inline requirements.yml:50]: Please also add this to `galaxy.yml` under `dependencies`

**Review by lewisdenny — CHANGES_REQUESTED** [review]

File: requirements.yml (hunk 3 above)
Comment (rebtoor) [reply requirements.yml:50]: done!

Comment (rebtoor) [reply requirements.yml]: I spoke too soon...

I'm using a single role because I don't need the entire collection but in the `dependency` section of galaxy.yml you can only define collections.  Well, since linux-system-roles don't have a single repo to handle the collection (they rely on a [script](https://github.com/linux-system-roles/auto-maintenance#release_collectionpy) for creating the release) so in any case i need to define it in its "galaxy" format.

And _obviously_ collection version is different from role version.

cc @pablintino @cjeanner

Comment (lewisdenny) [reply requirements.yml]: Ugh, supper annoying!

What you have done seems right to me, install the role as that's what we use and as we can't define a dependency to the single role we use, define it to `linux-system-roles` as it's still correct.

**Review by lewisdenny — APPROVED** [review]
/approve

File: docs/source/usage/01_usage.md [hunk 4]
```diff
@@ -68,6 +68,7 @@ are shared among multiple roles:
 - `cifmw_nolog`: (Bool) Toggle `no_log` value for selected tasks. Defaults to `true` (hiding those logs by default).
 - `cifmw_parent_scenario`: (String or List(String)) path to existing scenario/parameter file to inherit from.
 - `cifmw_configure_switches`: (Bool) Specifies whether switches should be configured. Computes in `reproducer.yml` playbook. Defaults to `false`.
+- `cifmw_ntp_server`: (String) Specifies an ntp server to use. Now it's only defined in `controller-0` environment. Defaults to `pool.ntp.org`.
```
Comment (cjeanner) [inline docs/source/usage/01_usage.md:71]: I'd say this:
```suggestion
- `cifmw_ntp_server`: (String) Specifies an ntp server to use. Now it's only used in `controller-0` environment. Defaults to `pool.ntp.org`.
```

File: galaxy.yml [hunk 5]
```diff
@@ -87,6 +87,7 @@ dependencies:
   'git+https://github.com/ansible-collections/mellanox.onyx': 'master'
   'git+https://github.com/openshift/community.okd': 'main'
   'git+https://github.com/ovirt/ovirt-ansible-collection': 'master'
+  'fedora.linux_system_roles': '1.87.1'
```
Comment (cjeanner) [inline galaxy.yml:90]: cant we use github to avoid ansible galaxy usual hiccups? That one is used whenever we run `ansible-galaxy collection install . -f` or related.

File: roles/reproducer/molecule/crc_layout/requirements.yml
Comment (cjeanner) [inline roles/reproducer/molecule/crc_layout/requirements.yml:1]: is this one really needed?

File: galaxy.yml (hunk 5 above)
Comment (rebtoor) [reply galaxy.yml:90]: we can't, i've explained why in a comment above. There's no official repo for the collection.

File: docs/source/usage/01_usage.md (hunk 4 above)
Comment (rebtoor) [reply docs/source/usage/01_usage.md:71]: ack!

File: roles/reproducer/molecule/crc_layout/requirements.yml
Comment (rebtoor) [reply roles/reproducer/molecule/crc_layout/requirements.yml:1]: yes, it's an external dependency that you need to install.

```
2024-08-20 07:39:44.930804 | TASK [Run molecule]
2024-08-20 03:39:46.330454 | controller | INFO     Found config file /home/zuul/src/github.com/openstack-k8s-operators/ci-framework/.config/molecule/config.yml
2024-08-20 03:39:46.668657 | controller | INFO     Performing prerun with role_name_check=0...
2024-08-20 03:40:03.293322 | controller | WARNING  Multiple versions of '@NAMESPACE@.@NAME@' were found installed, only the first one will be used, 3.2.1 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.294470 | controller | WARNING  Multiple versions of 'ansible.netcommon' were found installed, only the first one will be used, 7.0.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.295641 | controller | WARNING  Multiple versions of 'ansible.utils' were found installed, only the first one will be used, 5.1.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.297027 | controller | WARNING  Multiple versions of 'cifmw.general' were found installed, only the first one will be used, 1.0.0+1dfa9615 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.298171 | controller | WARNING  Multiple versions of 'cisco.ios' were found installed, only the first one will be used, 0.0.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.299493 | controller | WARNING  Multiple versions of 'community.crypto' were found installed, only the first one will be used, 2.22.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.300963 | controller | WARNING  Multiple versions of 'community.libvirt' were found installed, only the first one will be used, 1.3.1 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.302123 | controller | WARNING  Multiple versions of 'community.okd' were found installed, only the first one will be used, 4.0.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.303106 | controller | WARNING  Multiple versions of 'junipernetworks.junos' were found installed, only the first one will be used, * (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.304201 | controller | WARNING  Multiple versions of 'kubernetes.core' were found installed, only the first one will be used, 5.0.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.305586 | controller | WARNING  Multiple versions of 'mellanox.onyx' were found installed, only the first one will be used, 1.0.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.306953 | controller | WARNING  Multiple versions of 'openstack.cloud' were found installed, only the first one will be used, 2.2.0 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.308374 | controller | WARNING  Multiple versions of 'openstack.config_template' were found installed, only the first one will be used, 2.1.1 (/home/zuul/.ansible/collections/ansible_collections).
2024-08-20 03:40:03.335581 | controller | INFO     Running crc_layout > dependency
2024-08-20 03:40:04.607544 | controller | Starting galaxy role install process
2024-08-20 03:40:04.607609 | controller | - downloading role 'timesync', owned by linux-system-roles
2024-08-20 03:40:04.607985 | controller | - downloading role from https://github.com/linux-system-roles/timesync/archive/1.9.0.tar.gz
2024-08-20 03:40:04.608337 | controller | - extracting linux-system-roles.timesync to /home/zuul/.cache/molecule/reproducer/crc_layout/roles/linux-system-roles.timesync
2024-08-20 03:40:04.608774 | controller | - linux-system-roles.timesync (1.9.0) was installed successfully
2024-08-20 03:40:04.657248 | controller | INFO     Dependency completed successfully.
2024-08-20 03:40:04.658276 | controller | WARNING  Skipping, missing the requirements file.
2024-08-20 03:40:04.683644 | controller | INFO     Running crc_layout > prepare
```

File: roles/reproducer/tasks/configure_controller.yml [hunk 6]
```diff
@@ -472,3 +472,14 @@
       register: _sync_dep_install_result
       until: _sync_dep_install_result.finished
       retries: 20
+
+    - name: Configure the NTP service
```
[the remark this answers is not in the document]
Comment (pablintino) [reply roles/reproducer/tasks/configure_controller.yml:476]: Oh, true, I'm fine with it in that case.

File: requirements.yml [hunk 7]
```diff
@@ -13,6 +13,9 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
+roles:
+  - name: https://github.com/linux-system-roles/timesync.git
```
[the remark this answers is not in the document]
Comment (rebtoor) [reply requirements.yml:17]: From that exact page:

```
# from GitHub
- src: https://github.com/bennojoy/nginx
```

And probably the reason is [here](https://github.com/ansible/ansible/blob/69fb629355823e6f909ae21ee7bd068d17d3f6e9/lib/ansible/playbook/role/requirement.py#L101-L103)

tl;dr: with github.com url, it add prefix `git+` by itself. I can add `git+` for coherence though.

[the remark this answers is not in the document]
Comment (pablintino) [reply requirements.yml:17]: Oh, good catch, I searched inside the code but I didn't find that one. What you prefer. If it was me I'd add git+https just for consistency.

[the remark this answers is not in the document]
Comment (rebtoor) [reply requirements.yml:17]: let's leave it as-is for now, as for our chat discussion, the entire dependency topic should be optimized.

File: roles/reproducer/tasks/configure_controller.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (lewisdenny) [reply roles/reproducer/tasks/configure_controller.yml:476]: ansible--

**Comment by rebtoor** [comment]
/hold

back on dev, i found a corner case :)

**Comment by rebtoor** [comment]
/hold cancel

fixed

**Comment by raukadah** [comment]
from trigger hci job: https://sf.hosted.upshift.rdu2.redhat.com/logs/86/54186/1/check-rdo/ci-framework-baremetal-static-node-rhel9-va-hci/02e6bcb/job-output.txt
```
TASK [Include timesync role name=linux-system-roles.timesync] ******************
2024-08-27 09:46:44.716679 | controller | Tuesday 27 August 2024  09:46:44 -0400 (0:00:16.076)       1:14:07.260 ********
2024-08-27 09:46:44.716689 | controller | ERROR! the role 'linux-system-roles.timesync' was not found in /home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles:/home/zuul/ci-framework-data/artifacts/roles:/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/zuul/.ansible/roles:/home/zuul/src/github.com/openstack-k8s-operators/ci-framework
2024-08-27 09:46:44.722117 | controller |
2024-08-27 09:46:44.722155 | controller | The error appears to be in '/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles/reproducer/tasks/configure_controller.yml': line 490, column 19, but may
2024-08-27 09:46:44.722167 | controller | be elsewhere in the file depending on the exact syntax problem.
2024-08-27 09:46:44.722174 | controller |
2024-08-27 09:46:44.722180 | controller | The offending line appears to be:
2024-08-27 09:46:44.722187 | controller |
2024-08-27 09:46:44.722194 | controller |           ansible.builtin.include_role:
2024-08-27 09:46:44.722200 | controller |             name: linux-system-roles.timesync
2024-08-27 09:46:44.722206 | controller |                   ^ here
2024-08-27 09:46:44.722225 | controller |
```
